zebra-network/src/address_book.rs

```diff
@@ -141,6 +155,8 @@ impl AddressBook {
         // and it gets replaced by `update_metrics` anyway.
         let (address_metrics_tx, _address_metrics_rx) = watch::channel(AddressMetrics::default());
 
+        // Avoid initiating outbound handshakes when max_connections_per_ip is 1.
+        let should_limit_outbound_conns_per_ip = max_connections_per_ip == 1;
```

Comment:
Currently we assert that `initial_peer_set_size` is greater than zero, should we do the same with `max_connections_per_ip`?

Or should we remove the assertions on both and just warn?

```suggestion
        let should_limit_outbound_conns_per_ip = max_connections_per_ip <= 1;
```

Reply:
I think we should remove the assertions on both, use the default values, and log a warning.